### ISSUE TITLE
Support for Pi-Hole v6

### ISSUE THREAD

**Issue by squawkingVFR** [badsgahhl/pihole-browser-extension#85]
**Is your feature request related to a problem? Please describe.**

Pi-Hole v6 is in development and publicly available. It involves a completely rewritten REST-based API and this extension will require an update to function.

**Describe the solution you'd like**

Support for Pi-Hole v6 by the time it is formally released, or soon thereafter.

**Describe alternatives you've considered**

There isn't an alternative if this extension will be supported on Pi-Hole v6.

**Additional context**

Thanks for all your hard work on this plugin. It's invaluable for people like me. I'm hoping you're looking at v6-development and planning for a migration to the new API!

**Comment by badsgahhl** [comment]
#86

**Comment by Skylinar** [comment]
Any news on the v6 support?

**Comment by badsgahhl** [comment]
Probably looking into it the next days.

**Comment by TioBundy** [comment]
Me too, so much interested in this extension for chrome. Please, update to get compatibility for v6.

**Comment by t-makaro** [comment]
Something about my pihole setup broke today (ah the perils of auto-updating scripts). I ended up reimaging, and I'm now on v6.

**Comment by TioBundy** [comment]
> Hoy algo falló en mi configuración de Pihole (ah, los peligros de las actualizaciones automáticas de scripts). Terminé por volver a crear la imagen y ahora estoy en la versión 6.

I am definitely learning based on hits that you have to read the update notes and not update so lightly. And less without backing up.

**Comment by eugef66** [comment]
Updated to pi-hole V6 and my extension for Firefox stopped working (as excepted).  Really appreciate an update soon.  Love this extension and use it all the time.

**Comment by AlfaJackal** [comment]
+1

**Comment by siborg** [comment]
no pressure though and take your time. we appreciate you providing these
extremely useful extensions for free and . take it easy and get it ready
when you feel like it.

Si.

**Comment by rivman** [comment]
Many thanks for this extension that we hope to see again working. Bless you.

**Comment by Guyverix** [comment]
I am not a real developer, but an Ops guy.  LMK if there is any assistance I can give you on this to lighten the load.

**Comment by jdrch** [comment]
Thinking out aloud: what are the odds we as users can pressure Pi-hole's devs to stop breaking the API with every major release? It's getting to the point that users have to switch apps every time because volunteer 3rd party app devs can't keep up.

**Comment by upD8R** [comment]
Indeed, adapting to such changes is effort. But V6 had its beta phase started almost a year ago, so this is not a sudden surprise. 
If such a project is done in sparetime, developers priorities can easily shift and this is fine. In the end we use something, which was most likely created for one person and then offered (for free) to others. No obligations here. 

And you would also need to look for alternatives if a project gets abandoned but it seems this is not the case here 😉

**Comment by badsgahhl** [comment]
https://github.com/badsgahhl/pihole-browser-extension/releases/tag/4.0.0

**Comment by upD8R** [comment]
Thanks very much indeed - will this show up in the webstore, too?

**Comment by badsgahhl** [comment]
> Thanks very much indeed - will this show up in the webstore, too?

After Google has approved it. Webstore is by far the slowest so it will take up to 5 days.

**Comment by eugef66** [comment]
Many thanks

**Comment by jdrch** [comment]
Thanks for the update, but I keep getting `Connection failed` on Firefox. I'm running Pi-hole v6 and using https://my.pi-hole.ip.address/admin and my web UI password. Any ideas what I'm doing wrong?

**Comment by badsgahhl** [comment]
Yea the new authentication is *special*. Rather than using stateless tokens its using a cookie session token hybrid mix. The usage of the admin dashboard & extension in parallel can do weird stuff as they share the same cookies. 

Currently I implemented the bare minimum to handle the session, but there could be stuff improved if these connection errors occur more often.

**Comment by jdrch** [comment]
@badsgahhl OK I see. Thanks for the explanation!

**Comment by jdrch** [comment]
@badsgahhl regarding your cookie session token hybrid statement: I tried setting the extension up on a 2nd machine and it connected only after I logged into the Pi-hole web interface in a separate tab of the same browser.

**Comment by badsgahhl** [comment]
There will be one patch to solve some auth related stuff. This will prevent the cookie dilemma and only uses the session the extension really knows.

**Comment by ImpetuousRacer** [comment]
It's updated on the Firefox store. Could you also push the update to the Microsoft Edge store. Thanks so much for this!
https://microsoftedge.microsoft.com/addons/detail/pihole-browser-extension/jbelapanpabbnchmaalboimjmgmbfjnm

**Comment by badsgahhl** [comment]
> It's updated on the Firefox & Google Chrome store. Could you also push the update to the Microsoft Edge store. Thanks so much for this! https://microsoftedge.microsoft.com/addons/detail/pihole-browser-extension/jbelapanpabbnchmaalboimjmgmbfjnm

It's already published and waiting for the approval for it's release. The edge store is even slower than Webstore and will take up to 7 working days.

**Comment by bilderlf** [comment]
Hallo,
ich nutze pihole v6 mit addon v4.0.2 mit google Chrome (neuste Version).

Das roten Symbol hatte früher dauerhaft den Status "On", jetzt ist meist nur die rote Beere, ich muss 1x drauf klicken, dann zeigt es "On" an, aber es verschwindet sporadisch wieder.

Warum?

**Comment by Skylinar** [comment]
Klingt nach einem bug. Mach ein Bug Ticket auf. 
Und fürs nächste mal wäre es schön die Frage unter einem englisch sprachigem Thread auch in englisch zu formulieren, sodass alle verfolgen können, worum es geht.

**Comment by bilderlf** [comment]
> Klingt nach einem Bug. Mach ein Bug Ticket auf. Und fürs nächste Mal wäre es schön, die Frage unter einem englischsprachigen Thread auch in englisch zu formulieren, sodass alle verfolgen können, worum es geht.

wtf. kann dein Browser keine Übersetzung? ich schreibe in der Sprache die ich will, wenn DU damit Probleme hast ist das ganz allein deine Sache

**Comment by atnetsolutions** [comment]
> Are you using 2FA?

Nope...

**Comment by Skylinar** [comment]
@atnetsolutions 

I installed the latest version on my Firefox on MacOs as well, seems to work fine. Are you sure that you are using the right configuration? Maybe as well a good idea to check in the UI if you are really using the newest plugin version: 
`PiHole Browser Extension: 4.0.2 `

In configuration only the pihole ip: http://192.168.xxx.xxx without /admin

If you are not using 2FA you should be able to use either your login password or a generated "app" password.

**Comment by jdrch** [comment]
> In configuration only the pihole ip: http://192.168.xxx.xxx without /admin

I still can't get this to work reliably. The 1st PC I got it to work on is now saying `Connection failed` with `http://` and `https://` Pi-hole IP addresses, with and without `/admin`, and with both the login password and app password.

**Comment by Zathras42** [comment]
> > In configuration only the pihole ip: http://192.168.xxx.xxx without /admin
> 
> I still can't get this to work reliably. The 1st PC I got it to work on is now saying `Connection failed` with `http://` and `https://` Pi-hole IP addresses, with and without `/admin`, and with both the login password and app password. I'm using Firefox Nightly x64 on Windows 11 24H2.

Clear the sessions, as per https://github.com/badsgahhl/pihole-browser-extension/issues/103

**Comment by jdrch** [comment]
> > > In configuration only the pihole ip: http://192.168.xxx.xxx without /admin
> > 
> > 
> > I still can't get this to work reliably. The 1st PC I got it to work on is now saying `Connection failed` with `http://` and `https://` Pi-hole IP addresses, with and without `/admin`, and with both the login password and app password. I'm using Firefox Nightly x64 on Windows 11 24H2.
> 
> Clear the sessions, as per [#103](https://github.com/badsgahhl/pihole-browser-extension/issues/103)

Only 1 session shows up here, and it seems to be the direct web UI one, because when I clear it, I'm logged out of the web UI. Doing so changes nothing, unfortunately.

**Comment by lithophone** [comment]
Try http://XXX.XXX.XXX.XXX:8080/admin/

Your port might also be 80 (instead of 8080, in this case)

**Comment by jdrch** [comment]
> Try http://XXX.XXX.XXX.XXX:8080/admin/
> 
> Your port might also be 80 (instead of 8080, in this case)

No luck with either. I've even tried https://pi.hole/admin/api/auth as the [API docs](https://pihole-docs.netlify.app/api/auth/) suggest, with no success.

**Comment by lithophone** [comment]
> > Try http://XXX.XXX.XXX.XXX:8080/admin/
> > Your port might also be 80 (instead of 8080, in this case)
> 
> No luck with either. I've even tried https://pi.hole/admin/api/auth as the [API docs](https://pihole-docs.netlify.app/api/auth/) suggest, with no success.

Have you been able to access the Pi-Hole interface directly through your browser?  If so, using what address?  I'm no professional by any means.  But in my case it was using the exact same browser URL for the PiHole Address in the browser extension.

**Comment by jdrch** [comment]
> > > Try http://XXX.XXX.XXX.XXX:8080/admin/
> > > Your port might also be 80 (instead of 8080, in this case)
> > 
> > 
> > No luck with either. I've even tried https://pi.hole/admin/api/auth as the [API docs](https://pihole-docs.netlify.app/api/auth/) suggest, with no success.
> 
> Have you been able to access the Pi-Hole interface directly through your browser? If so, using what address? I'm no professional by any means. But in my case it was using the exact same browser URL for the PiHole Address in the browser extension.

Yep, that's what I had originally and what was working until it suddenly stopped working today.

**Comment by lithophone** [comment]
So you CAN'T access directly via the browser (not using the extension?)  Are you sure the Pi-Hole device is still running?  Maybe restart?

**Comment by jdrch** [comment]
> So you CAN'T access directly via the browser (not using the extension?) Are you sure the Pi-Hole device is still running? Maybe restart?

Huh? Yes, I can access the web UI just fine. Using the same URL for that in the extension fails.

**Comment by lithophone** [comment]
Hmmm, sorry.  If you have the web UI URL working, I'd try (if you haven't already) the generic troubleshooting steps of removing the Pi-hole instance from the extension and readding.  Or even uninstalling the extension and reinstalling that.  Again, not an expert.  I know the URLs took some trial and error, and varied between what worked between this and the V6 Home Assistant integration.  Just offering suggestions.  Hope you can get it working.

**Comment by jdrch** [comment]
Tried everything and still no success. For others on Windows who are having the same problem, the Windows build of [Pi-hole client](https://github.com/tsutsu3/pi-hole-client works just fine as a workaround.

**Comment by Tekwyzard** [comment]
I've just been having a bit of a similar battle as others above, on one computer, windows 7 (yes, I know the risks), the Firefox updated add-on works fine, on the other, a newly set up raspberry pi5, fully up to date, the add-on was behaving as mentioned by other people. I tried all the above mentioned fixes, with no result. Mine won't accept anything but https connections for some reason, but whatever.
Out of desperation, instead of using https://xxx.xxx.xxx.xxx/admin/ on my Pi, my usual setting which does work on the windows machine, I tried https://pi.hole/admin/ and it worked fine. Much like others on here, I have no idea why there's a difference between how my two machines behave when given the same credentials to the one PiHole.
